ptx/src/test/spirv_run/mod.rs

```diff
@@ -224,6 +238,34 @@ fn test_hip_assert<
     Ok(())
 }
 
+fn test_llvm_assert<
+    'a,
+>(
+    name: &str,
+    ptx_text: &'a str,
+    expected_ll: &str
+) -> Result<(), Box<dyn error::Error + 'a>> {
+    let ast = ptx_parser::parse_module_checked(ptx_text).unwrap();
+    let llvm_ir = pass::to_llvm_module(ast).unwrap();
+    let actual_ll = llvm_ir.llvm_ir.print_as_asm();
+    let result = catch_unwind(||

```

Comment:
Hmmm, that catch_unwind is really unfortunate. I can see that crate `pretty_assertions` also exports `StrComparison` type which seem to wrap the presentation of the diff.
Please check if we can get the same effect and avoid touching unwinding with `if actual_ll != expected_ll { panic!("{}", pretty_assertions::StrComparison::new(actual_ll, expected_ll)); }`